diff --git a/src/cache.cc b/src/cache.cc
index 85fa902..2620fd7 100644
--- a/src/cache.cc
+++ b/src/cache.cc
@@ -72,6 +72,7 @@ void BufferPoolManager::evict() {
         if (page) {
             disk_manager.write_page(victim_id, page);
             delete page;
+            page_directory.erase(it);
             Logger::get_instance().warn("Cache", "Evicted page " + std::to_string(victim_id) + " from buffer pool.");
         }
     }
